### PR TITLE
🐛 Fix byte-to-character position conversion in PGliteInstanceManager

### PR DESCRIPTION
**Pull request by hoshinotsuyoshi**
## Issue

- resolve: https://github.com/route06/liam-internal/issues/5667

## Why is this change needed?

The PostgreSQL parser returns byte positions for SQL statements, but JavaScript's string methods use character positions. This mismatch caused statement extraction to fail with multi-byte characters like U+2019 (right single quotation mark) and emojis, resulting in truncated SQL statements (e.g., "COMMENT" becoming "OMMENT").

## Summary

This PR fixes the byte-to-character position conversion in PGliteInstanceManager to properly handle multi-byte characters including emojis and Unicode characters.

### Changes Made

1. **Fixed `byteToCharPosition()` method**:
   - Now correctly handles surrogate pairs and multi-byte characters
   - Uses string iterator (`for...of`) to properly count characters
   - Accurately converts byte positions to character positions

2. **Added comprehensive test cases**:
   - Tests for U+2019 (right single quotation mark) handling
   - Tests for escaped apostrophes (`''`) in SQL strings  
   - Tests for emojis (📝🚀✅💡) in COMMENT statements
   - Tests for mixed Japanese, Chinese, and emoji characters

### Technical Details

The issue occurred because:
- PostgreSQL parser (`@pgsql/parser`) returns `stmt_location` in bytes
- JavaScript's `String.slice()` expects character positions
- Multi-byte characters (emojis: 3-4 bytes, CJK: 3 bytes) caused position misalignment

The fix ensures all multi-byte characters are properly handled when extracting individual SQL statements from parsed results.

## Test Results

All tests pass including:
- Existing statement parsing tests
- New multi-byte character tests
- Emoji and international character tests

## Related Links

- Original issue: https://github.com/route06/liam-internal/issues/5667
- Related PR (incomplete fix): #3492

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixes SQL extraction so COMMENT ON and other statements with emojis, multi‑byte characters, and various apostrophes (including U+2019) are not truncated; preserves full content and yields accurate parse behavior for stmt_location edge cases.

* **Tests**
  * Adds comprehensive tests covering escaped apostrophes, U+2019, surrogate pairs/emojis, mixed languages, and stmt_location edge cases (note: test suite appears duplicated; one test is skipped for a known issue).

### CODE REVIEW

File: frontend/internal-packages/pglite-server/src/PGliteInstanceManager.test.ts [hunk 1]
```diff
@@ -300,4 +300,108 @@ describe('PGliteInstanceManager', () => {
...
+        r.sql.includes('COMMENT ON'),
+      )
+      expect(commentStatements).toHaveLength(8)
+
+      commentStatements.forEach((result) => {
+        expect(result.sql.trim()).toMatch(/^COMMENT ON/)
+        // Should not match truncated patterns
+        expect(result.sql).not.toMatch(/^OMMENT/)
+        expect(result.sql).not.toMatch(/^MMENT/)
+        expect(result.sql).not.toMatch(/^MENT/)
```
Comment (hoshinotsuyoshi) [inline frontend/internal-packages/pglite-server/src/PGliteInstanceManager.test.ts:362]: regression test
